### ISSUE TITLE
Feature request: return property keys of a JSValue

### ISSUE THREAD

**Issue by Symbitic** [justjake/quickjs-emscripten#185]
When a JSValue is a JavaScript object, I would like a way to return a list of all the valid properties that could be passed to `getProp`. This would help with error detection if an expected property isn't there and with iteration.

**Comment by justjake** [comment]
Agree!

In the mean time you can get away with this sort of thing:

```js
function keys(context, handle) {
  using objectHandle = context.getProp(context.global, 'Object')
  using keysFnHandle = context.getProp(objectHandle, 'keys')
  using keysHandle = context.callFunction(keysFnHandle, context.undefined, handle)
  return context.dump(keysHandle)
}
```

You can move fetching the `Object.keys` part out of the function so you do it only once.

**Comment by justjake** [comment]
Happy to accept a pull request that adds a lower-level function using an iterator, etc.

**Comment by justjake** [comment]
This is published as of 0.30.0
